File: solutions/Java/longest_substring_without_repeating_characters_brute_force.java
import java.util.HashSet;

public class Main {
    public static int longestSubstringWithoutRepeatingCharactersBruteForce(String s) {
        int maxLength = 0;

        for (int i = 0; i < s.length(); i++) {
            HashSet<Character> charSet = new HashSet<>();
            for (int j = i; j < s.length(); j++) {
                if (charSet.contains(s.charAt(j))) {
                    break;
                }
                charSet.add(s.charAt(j));
                maxLength = Math.max(maxLength, j - i + 1);
            }
        }

        return maxLength;
    }

    public static void main(String[] args) {
        String s = "abcabcbb";
        System.out.println(longestSubstringWithoutRepeatingCharactersBruteForce(s));
    }
}

File: solutions/Python/longest_substring_without_repeating_characters_brute_force.py
def longest_substring_without_repeating_characters_brute_force(s: str) -> int:
    max_length = 0

    for i in range(len(s)):
        char_set = set()
        for j in range(i, len(s)):
            if s[j] in char_set:
                break
            char_set.add(s[j])
            max_length = max(max_length, j - i + 1)

    return max_length

print(longest_substring_without_repeating_characters_brute_force("abcabcbb"))  # 3
